web_src/js/features/repo-diff.js

```diff
@@ -8,6 +8,24 @@ const {csrfToken} = window.config;
 export function initRepoDiffReviewButton() {
   $(document).on('click', 'button[name="is_review"]', (e) => {
     $(e.target).closest('form').append('<input type="hidden" name="is_review" value="true">');
+
+    // Watch for the form's submit event.
+    e.target.closest('form').addEventListener('submit', () => {
+      // Set a zero-timeout, so this would be executed after the network request has finished.
+      setTimeout(() => {
+        const commentCounter = document.querySelector('#review-box .review-comments-counter');
+        // Remove the display: none.
+        commentCounter.style.display = '';
+        // Increase counter by one, in case it's the first review, parseInt will return `NaN`
+        // and default back to '1'.
+        commentCounter.textContent = parseInt(commentCounter.textContent) + 1 || '1';
+
+        // Make the review-box to do a little pulse.
+        document.querySelector('#review-box').classList.add('pulse');
+        // Remove the pulse class once it's done.
+        setTimeout(() => document.querySelector('#review-box').classList.remove('pulse'), 2500);
```

Comment:
Is there no way to get the duration of the animation?
Something like
```suggestion
        setTimeout(() => document.querySelector('#review-box').classList.remove('pulse'), document.querySelector('#review-box').style.animationDuration);
```
?
Because it is definitely safer if we compute that dynamically rather than absolute as it could change over time.

Reply:
I'd store it in a CSS variable and then retrieve it via `getComputedStyle` like done [here](https://github.com/go-gitea/gitea/blob/7b089c465d9f851454be91ca209ab26782648a45/web_src/js/features/codeeditor.js#L71-L72). It could be made a function in `utils.js`:

````js
export function getCSSVariable(name) {
  const styles = window.getComputedStyle(document.documentElement);
  return styles.getPropertyValue(name).trim();
}
````

Reply:
It's just one line in jQuery:  `$(...).css('animation-duration')`

Why do you like to re-invent wheels .................

Reply:
Actually, above suggestion may not perform so well because `getComputedStyle` is expensive , but `.style.animationDuration` certainly will (or the jQuery variant if you really must).

Reply:
We must use `getComputedVariable`, because we don't use CSS inline for the animation-duration property. But I have another trick up my sleeve, that avoids this weird way of getting the animation duration, but it as ~~slow~~fast as `getComputedStyle`.

Reply:
https://github.com/go-gitea/gitea/pull/19612/commits/f2e8d3a68bfffc13664f33b685c9d5df129932c7